src/util.cc

```diff
@@ -1017,10 +1021,15 @@ ssize_t parse_content_disposition(char* dest, size_t destlen,
         state = CD_AFTER_VALUE;
       }
       else {
-        /* TEXT which is OCTET except CTLs, but including LWS. We only
-           accept ISO-8859-1 chars. */
+        /* TEXT which is OCTET except CTLs, but including LWS. accept
+           ISO-8859-1 chars, or UTF-8 if content-disposition-default-utf8 is set */
         quoted_seen = 0;
-        if (!isIso8859p1(*p)) {
+        if (default_utf8){
+          if (utf8dfa(&dfa_state, &dfa_code, (unsigned char)*p) == UTF8_REJECT) {
+            return -1;
```

Comment:
This may leave utf-8 string which is prematurely truncated.  dfa_state must be verified that it is UTF8_ACCEPT in somewhere, for example, in CD_AFTER_VALUE.